consensus/ibft/consensus_backend.go

```diff
@@ -64,6 +70,21 @@ func (i *backendIBFT) InsertBlock(
 		return
 	}
 
+	// WriteCommittedSeals alters the extra data before writing the block
+	// It doesn't handle errors while pushing changes which can result in
+	// corrupted extra data.
+	// We don't know exact circumstance of the unmarshalRLP error
+	// This is a safety net to help us narrow down and also recover before
+	// writing the block
+	if err := i.ValidateExtraDataFormat(newBlock.Header); err != nil {
+		i.logger.Error("cannot write block: corrupted extra data",
+			"err", err,
+			"\n\tbefore", hex.EncodeToHex(extraDataBackup),
+			"\n\tafter", hex.EncodeToHex(header.ExtraData))
```

Comment:
@trimixlover i suggest to print the committed seals as received by consensus here as well to have everything needed to debug.